libs/portal-integration-angular/testing/diagram.harness.ts

```diff
@@ -0,0 +1,16 @@
+import { ComponentHarness } from '@angular/cdk/testing'
+import { DivHarness } from './div.harness'
+
+export class DiagramHarness extends ComponentHarness {
+  static hostSelector = 'ocx-diagram'
+
+  getSumKeyNumber = this.locatorForOptional(DivHarness.with({ class: 'sumKey' }))
+
+  async getSumKey(): Promise<string | null> {
+    return await (await this.host()).getAttribute('ng-reflect-sum-key')
```

Comment:
Please don't use the internal ng attributes here. This is not a test and is highly unstable. Also, you are not interested in the key but in the text so you can get it from the div